### PR TITLE
test: Convert mutation explain tests to new explain setup

### LINKED ISSUE #1379: Convert basic default explain tests including filter tests.
Convert basic default explain tests including filter tests.

More info in parent ticket: https://github.com/sourcenetwork/defradb/issues/953

### LINKED ISSUE #1417: Convert default mutation explain tests
Convert default mutation explain tests.

More info in parent ticket: https://github.com/sourcenetwork/defradb/issues/953

### PR DESCRIPTION
**Pull request by shahzadlone**
## Relevant issue(s)
- Part of #953
- Resolves #1417

## Description
Continue converting explain tests to the new explain setup before we can integrate the entire setup to the new action based setup. #953 Has a lot more detail on the entire plan.

This PR converts all the default mutation explain tests to the new explain setup.

- Splits delete tests into two files one with error and without.
- Splits multi tests from single function into separate test functions (easier to maintain and debug). 
- Removes some tests that provide no value wrt to non-execution explain tests.

### CODE REVIEW

File: tests/integration/explain/default/create_test.go [hunk 1]
```diff
@@ -80,32 +79,19 @@ func TestExplainMutationCreateSimpleDoesNotCreateDocGivenDuplicate(t *testing.T)
 			},
 		},
 
-		Results: []dataMap{
+		ExpectedTargets: []explainUtils.PlanNodeTargetCase{
```
Comment (fredcarle) [inline tests/integration/explain/default/create_test.go:82]: question: It seams as if a lot of the changes to the tests, including this one, are testing less of the output. Am I missing something or is this the intent?

Comment (shahzadlone) [reply tests/integration/explain/default/create_test.go:82]: It's now testing the pattern (ordering of the nodes), and only the content of the nodes that we care about for each test separately. For this specific test for example we only care about the `createNode`'s attributes (point 2 from the below link).

More info can be found here:
https://github.com/sourcenetwork/defradb/pull/949#issuecomment-1327719247

**Review by AndrewSisley — APPROVED** [review]
LGTM

File: tests/integration/explain/default/create_test.go (hunk 1 above)
Comment (fredcarle) [reply tests/integration/explain/default/create_test.go:82]: Thanks for referencing that comment! It makes much more sense to me now :)

**Review by fredcarle — APPROVED** [review]
LGTM :)

**Pull request merged**